crates/runtime/tests/models/openai.rs

```diff
@@ -390,7 +457,56 @@ async fn openai_test_chat_messages() -> Result<(), anyhow::Error> {
     // ai_completion input message is deterministic - based on available tools, app configuration, and the input message
     insta::assert_snapshot!(
         "chat_1_ai_completion_input",
-        format!("{:?}", ai_completion_task.1)
+        pretty_json_str(&ai_completion_task.1)?
+    );
+
+    // Test 2: Chat completion related to searching for items (similirity search)
+
+    let req = CreateChatCompletionRequestArgs::default()
```

Comment:
It might be worth having one inlined function per test case. e.g.
```rust
#[tokio::test]
#[allow(clippy::too_many_lines)]
async fn openai_test_chat_messages() -> Result<(), anyhow::Error> {
        // Setup everything

        async fn test_foo() {

        }

        async fn test_bar() {

        }
      
        test_foo().await;
        test_bar().await;
        
        // Teardown everything
}
```
I don't know of a more axiomatic way in `tokio::test` to do this.

Reply:
Thank you - added to helper functions for sql_query and similarity_search tests:

```rust
let model_secrets = get_params_with_secrets(&model_with_tools.params, &rt).await;
let tool_model = try_to_chat_model(&model_with_tools, &model_secrets, Arc::clone(&rt)).await?;

verify_sql_query_chat_completion(&*tool_model, &rt, &trace_provider).await?;
verify_similarity_search_chat_completion(&*tool_model, &rt, &trace_provider).await?;

Ok(())
```